Swift/Errors.swift

```diff
@@ -0,0 +1,237 @@
+//
+//  Errors.swift
+//  CBL ObjC
+//
+//  Copyright (c) 2021 Couchbase  Inc All rights reserved.
+//
+//  Licensed under the Apache License  Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing  software
+//  distributed under the License is distributed on an "AS IS" BASIS
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND  either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+
+import Foundation
+
+public struct CBLError {
+    public static let domain = CBLErrorDomain
+    
+    /// Internal assertion failure
+    public static let assertionFailed       = 1
```

Comment:
For easy reference for the users, declaring the values here seems to be better. 

Else users might want to go and see the objc documentation together with swift in order to understand the error code.